### DISCUSSION TITLE
To have a higher order element for displacement in coupled problems.

### DISCUSSION THREAD

**Discussion by Khaksarkn** [idaholab/moose#26999] [Q&A General] [answered]
Hi, 
I have a THM problem containing Temperature, Pressure, And Displacements. 
I was wondering how I could have second-order elements for displacements while keeping first-order for pressure and temperature.
I use Cubit and the Mesh generator of MOOSE. 
I used the ORDER feature for the elements but it gave me this error. I am not sure how to specify higher-order elements in the mesh.

![image](https://github.com/idaholab/moose/assets/77813275/ff1b2a72-1308-4c39-bd97-b4bb2d4d68da)

This is the input file:

[Mesh]
  type = GeneratedMesh
  dim = 2
  nx = 2
  ny = 20
  xmin = -0.1
  xmax = 0.1
  ymin = -2
  ymax = 0
[]

[Variables]
  [temperature]
    initial_condition = 274.15
    family = LAGRANGE
    order = FIRST
    scaling = 1e3
  []
  [porePressure]
    family = LAGRANGE
    order = FIRST
    scaling = 1e9
  []

  [disp_x]
    family = LAGRANGE
    order = SECOND
  []
  [disp_y]
    family = LAGRANGE
    order = SECOND
  []
[]

I would appreciate your help. 
Thanks.

**Comment by GiudGiud** [comment]
Hello

On a regular MOOSE simulation, this would have been possible.
In THM, it's additional development work if you want to do that. It should not be impossible, @joshuahansel can comment on the expected difficulty

Guillaume

**Reply by lindsayad** [reply, marked as answer]
In order to combat this particular error, you can put `second_order = true` in your `Mesh` block

**Reply by GiudGiud** [reply]
ok I misunderstood completely the problem.

For THM simulation you want to add `2nd_order_mesh = true` to the GlobalParams block (I think it on a per component basis)
You probably dont have a Mesh block

**Reply by Khaksarkn** [reply]
Thank you, Alex and Guillaume!
Yes, just setting ```second_order = true``` in the `mesh` block was the problem. 
I appreciate it.
